generators/dart.js

```diff
@@ -13,14 +13,32 @@
 import * as goog from '../closure/goog/goog.js';
 goog.declareModuleId('Blockly.Dart.all');
 
-import './dart/colour.js';
-import './dart/lists.js';
-import './dart/logic.js';
-import './dart/loops.js';
-import './dart/math.js';
-import './dart/procedures.js';
-import './dart/text.js';
-import './dart/variables.js';
-import './dart/variables_dynamic.js';
+import {DartGenerator} from './dart/dart_generator.js';
+import * as colour from './dart/colour.js';
+import * as lists from './dart/lists.js';
+import * as logic from './dart/logic.js';
+import * as loops from './dart/loops.js';
+import * as math from './dart/math.js';
+import * as procedures from './dart/procedures.js';
+import * as text from './dart/text.js';
+import * as variables from './dart/variables.js';
+import * as variablesDynamic from './dart/variables_dynamic.js';
 
 export * from './dart/dart_generator.js';
+
+/**
+ * Dart code generator instance.
+ * @type {!DartGenerator}
+ */
+export const dartGenerator = new DartGenerator();
+
+// Add reserved words.  This list should include all words mentioned
+// in RESERVED WORDS: comments in the imports above.
+dartGenerator.addReservedWords('Html,Math');
```

Comment:
Could you have the modules export arrays of reserved words? I imagine this would break the `Object.assign` below, but wanted to mention in case there was a way to work around that.

Reply:
I actually thought about that.  It's doable: a particular export can be pulled out using a destructuring assignment, leaving the rest on a remainder object which could then be passed to `assign`.

I also looked into whether ES modules can have symbol-keyed exports.  Regular exports just become ordineary string-keyed properties on the `Module` object, so why not provide a way to have properties on the `Module` object that have a `Symbol` as their key?  (They in fact already have a property keyed by [Symbol.toStringTag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) so that they get reported as `[object Module]` by `Object.prototype.toString()`.)

Alas, it turns out that that there's no way to do that, and, due to the absence of symbol literals (symbols only really exist at runtime) combined with the way that exports are statically analysed, not really any good way for a future ES202x to make it possible.

I had another thought, though: what about putting an extra property on the function objects themselves?  That way each individual one carries with it the information needed to use it.  This could be scanned by `generator/<lang>.js` at installation time, or even in the `<Lang>Generator`'s `init` method.

Opinions welcome, though I'm inclined to leave implementing any of these options (well, except for the infeasible one) for a later PR, though, unless you think the current code unacceptable.

Reply:
Agree! best to leave for a separate PR. I'll respond to the rest of your questions laterz, just wanted you to know this isn't blocking =)

Reply:
+1 that finding a way to export the reserved words would be useful in a follow-up. Putting an extra property on the function object makes me feel weird, but it would be nice if people who are importing individual functions couldn't forget this part.

Reply:
> it would be nice if people who are importing individual functions couldn't forget this part.

🤔 We could include the reserved words in the name of the export:
```
export function math_arithmetic$RESERVED$Math(block, generator) {/* ... */}
```
😈

Filed #7187 for further discussion.